posts/deploying-a-jhipster-app-to-aws-using-elastic-beanstalk.md

```diff
@@ -0,0 +1,101 @@
+---
+
+[JHipster](https://www.jhipster.tech/) is a great development platform to help you bootstrap a modern Web app in less than 5 minutes. One small detail is that everything runs on your local laptop which can be an issue if you want to showcase your app to external people. Cloud providers like [AWS](https://aws.amazon.com/), [Azure](https://azure.microsoft.com) or [GCP](https://cloud.google.com/) are the current leaders and they provide services to help you deploy your JHipster app to the Cloud.
+
+In this blog, I will focus on AWS and especially on Elastic Beanstalk which is an orchestration service. Elastic Beanstalk helps you create a production-ready infrastructure that has the architecture below:
+
+![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/diagram-eb.png)
+
+# App generation with JHipster
+
+Currently the Elastic Beanstalk sub-generator only supports SQL databases (Oracle and Microsoft SQL Server excluded). Our JHipster app will be a basic monolith that uses React and PostgreSQL as a database.
+
+```json
+{
+  "generator-jhipster": {
+    "promptValues": {
+      "packageName": "com.company",
+      "nativeLanguage": "en"
+    },
+    "jhipsterVersion": "6.4.1",
+    "applicationType": "monolith",
+    "baseName": "jhipster",
+    "packageName": "com.company",
+    "packageFolder": "com/company",
+    "serverPort": "8080",
+    "authenticationType": "jwt",
+    "cacheProvider": "ehcache",
+    "enableHibernateCache": true,
+    "websocket": false,
+    "databaseType": "sql",
+    "devDatabaseType": "h2Disk",
+    "prodDatabaseType": "postgresql",
+    "searchEngine": false,
+    "messageBroker": false,
+    "serviceDiscoveryType": false,
+    "buildTool": "maven",
+    "enableSwaggerCodegen": false,
+    "embeddableLaunchScript": false,
+    "useSass": true,
+    "clientPackageManager": "npm",
+    "clientFramework": "react",
+    "clientTheme": "none",
+    "clientThemeVariant": "",
+    "testFrameworks": [],
+    "jhiPrefix": "jhi",
+    "entitySuffix": "",
+    "dtoSuffix": "DTO",
+    "otherModules": [],
+    "enableTranslation": true,
+    "nativeLanguage": "en",
+    "languages": [
+      "en"
+    ],
+    "blueprints": []
+  }
+}
+```
+
+The `.yo-rc.json` above can be used to avoid answering all the questions manually.
+
+# Deployment to AWS with Elastic Beanstalk
+
+ First, make sure that your AWS CLI is [correctly configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration) and the account must have the `AWSElasticBeanstalkFullAccess` policy. If you use a named profile different than `default`, you just have to set the environment variable `AWS_PROFILE` with the correct profile.
+
+ The command `jhipster aws` will ask questions about the AWS deployment:
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/jhipster-aws.png)
```

Comment:
To avoid confusion, have a graphic with more than just `jhipster` filled in for every question.